pkg/cloudprovider/providers/gce/gce_util.go

```diff
@@ -100,3 +100,25 @@ func isHTTPErrorCode(err error, code int) bool {
 	apiErr, ok := err.(*googleapi.Error)
 	return ok && apiErr.Code == code
 }
+
+// splitProviderID splits a provider's id into core components.
+// A providerID is build out of '${ProviderName}://${project-id}/${zone}/${instance-name}'
+// See cloudprovider.GetInstanceProviderID.
+func splitProviderID(providerID string) (project, zone, instance string, err error) {
```

Comment:
It is doable with `url.Parse()` but it still would be an equal amount of complexity.
You couldn't tell apart empty parameters for example in `gce:///foo/bar`.
You however can use it to get the Scheme of the URI, but afterwards you still have to split the path and access `URL.Host`.